wait-for.c

```diff
@@ -245,21 +243,21 @@ int main(int argc, const char **argv) {
 		goto exit;
 	}
 
-	if (args.username == NULL) {
-		// attempt to populate based on current UID
-		struct passwd *pwd = getpwuid(getuid());
-		if (pwd == NULL) {
+	errno = 0;
+	if (args.username) {
+		pwd = getpwnam(args.username);
+	} else {
+		pwd = getpwuid(getuid());
+	}
+
+	if (pwd == NULL) {
+		if (errno) {
 			perror("could not get passwd entry for the user");
 			status = 1;
```

Comment:
User `printf` and show the username; use `strerror` to show the error message (search the source for an example if you need one)